tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchReindexingTests.java

```diff
@@ -178,8 +179,112 @@ public void testRecreateIndex() throws Exception
 
         // TIDY
         // Restart ElasticsearchConnector.
+        cleanUpIndex();
+        AlfrescoStackInitializer.liveIndexer.start();
+
+    }
+
+    /**
+     * Common testing method for reindexing enabled and disabled features tests.
+     * @param metadataIndexingEnabled Reindexing metadata is enabled when true, disabled when false
+     * @param contentIndexingEnabled Reindexing content is enabled when true, disabled when false
+     * @param pathIndexingEnabled Reindexing path is enabled when true, disabled when false
+     * @param queryString Verification query string. It may include a <DOCUMENT_NAME> mark that is replaced by the actual document name created.
+     * @param expectingDocNameAsResult Result from verification query string is the name of the document created when true, empty result when false.
+     */
+    private void internalTestEnabledFeatures(
+        Boolean metadataIndexingEnabled,
+        Boolean contentIndexingEnabled,
+        Boolean pathIndexingEnabled,
+        String queryString,
+        Boolean expectingDocNameAsResult
+    ) throws Exception
+    {
+
+        // Initial timestamp for reindexing by date: this will save reindexing time for these tests
+        ZonedDateTime now = ZonedDateTime.now(Clock.systemUTC());
+        String testStart = DateTimeFormatter.ofPattern("yyyyMMddHHmm").format(now.minusMinutes(1));
+
+        // GIVEN
+        // Stop ElasticsearchConnector
+        AlfrescoStackInitializer.liveIndexer.stop();
+        // Create document
+        String documentName = createDocument();
+        // Delete index documents
+        cleanUpIndex();
```

Comment:
Do we need to do this?

Reply:
Is it better to create a single document for every tests?
This approach is creating a new document for each test and cleaning the existing Elasticsearch index to be sure that only the new document is target of the reindexing process.

Reply:
But do we mind if the reindexing process affects more nodes?